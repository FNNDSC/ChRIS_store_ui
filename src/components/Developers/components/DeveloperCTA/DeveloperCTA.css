.developer-cta {
  /* color */
  background: #393f44;
  background: var(--pf-black-800);
  background: linear-gradient(135deg, #4d5258, #393f44);
  background: linear-gradient(135deg, var(--pf-black-700), var(--pf-black-800));
  background-image: url('../../../../assets/img/banner_chris.png');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;

  /* display */
  padding: 20px;
}

.developer-cta-overview, .developer-cta-form { margin: 15px; }

.developer-cta-overview { flex: 3; }

.developer-cta-form { flex: 2; }

.developer-cta-header, .developer-cta-desc {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);
}

.developer-cta-header {
  font-weight: 200;
  font-size: 3.4em;
  line-height: 1.3;
}

.developer-cta-desc {
  /* display */
  margin-top: 20px;

  /* font */
  font-weight: 600;
  font-size: medium;
  text-shadow: 3px 3px 6px #030303;
  text-shadow: 3px 3px 6px var(--pf-black);
}

/* responsive breakpoints */
@media (min-width: 480px)  {
  .developer-cta { padding: 4em; }
}
