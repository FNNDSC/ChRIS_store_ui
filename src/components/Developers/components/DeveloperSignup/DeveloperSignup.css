#developer-signup-form {
    color: black;
}

#developer-signup-creating {
    vertical-align: super;
}

#developer-signup-creating {
    vertical-align: super;
}

.pf-c-card__body:hover {
    background-color: #ffffff;
}
