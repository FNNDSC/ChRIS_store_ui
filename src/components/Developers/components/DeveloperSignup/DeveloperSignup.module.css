.developer-signup-creating {
    vertical-align: super;
}
