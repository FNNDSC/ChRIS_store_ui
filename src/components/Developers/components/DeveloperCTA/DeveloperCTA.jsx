import React from 'react';
import PropTypes from 'prop-types';

import './DeveloperCTA.css';

import { Card, CardBody, Grid, GridItem, Bullseye } from '@patternfly/react-core';
import ConnectedDeveloperSignup from '../DeveloperSignup/DeveloperSignup';
import ChrisStore from '../../../../store/ChrisStore';

export const DeveloperCTA = ({ store }) => (
  <Bullseye id="developer-cta">
    <Grid style={{ maxWidth: '1200px' }} hasGutter>
      <GridItem lg={6} xs={12}>
        <div id="developer-cta-header">
          <h2>Expand the reach of your image processing software</h2>
          <p style={{ fontSize: 'medium' }}>
            A ChRIS Developer account enables you to submit your image
            processing application as a containerized ChRIS plugin and
            share it with the broader ChRIS community of researchers and
            clinicians.
          </p>
        </div>
      </GridItem>
      <GridItem lg={1} xs={12}/>
      <GridItem lg={5} xs={12}>
        {
          store.get('isLoggedIn') ? null :
            <div id="developer-cta-form">
              <Card>
                <CardBody>
                  <h2 style={{ margin: '0.25em 0 1.5em 0' }}><b>Create a ChRIS Developer Account</b></h2>
                  <ConnectedDeveloperSignup />
                </CardBody>
              </Card>
            </div>
        }
      </GridItem>
    </Grid>
  </Bullseye>
);

export default ChrisStore.withStore(DeveloperCTA);

DeveloperCTA.propTypes = {
  store: PropTypes.objectOf(PropTypes.object).isRequired,
};
