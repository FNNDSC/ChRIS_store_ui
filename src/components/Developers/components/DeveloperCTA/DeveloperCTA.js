import React from 'react';
import { Card, CardBody } from 'patternfly-react';
import './DeveloperCTA.css';
import DeveloperSignup from '../DeveloperSignup/DeveloperSignup';

const DeveloperCTA = () => (
  <div className="developer-cta">
    <div className="row developer-cta-container">
      <div className="developer-cta-overview">
        <div className="developer-cta-header">
          Expand the reach of your image processing software
        </div>
        <div className="developer-cta-desc">
          <p>
          ChRIS is an <strong>open source platform</strong> for medical
          analytics in the cloud, democratizing the development of image
          processing apps within an ecosystem following
            <strong>
              {' '}common standards, rather than disparate silos
            </strong>.
          </p>

          <p>
          A ChRIS Developer account enables you to submit your image
          processing application as a containerized ChRIS plugin and
          share it with the broader ChRIS community of researchers and
          clinicians. <strong>Join us!</strong>
          </p>
        </div>
      </div>
      <div className="developer-cta-form">
        <Card>
          <CardBody>
            <DeveloperSignup />
          </CardBody>
        </Card>
      </div>
    </div>
  </div>
);

export default DeveloperCTA;
