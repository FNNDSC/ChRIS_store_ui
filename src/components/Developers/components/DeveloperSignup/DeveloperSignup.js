import React, { Component } from 'react';
import {
  Form,
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock,
  Button,
  Alert,
  Spinner,
} from 'patternfly-react';
import StoreClient from '@fnndsc/chrisstoreapi';
import './DeveloperSignup.css';

/* inspired by http://bit.ly/2KycT4G */
const isTouchDevice = () => {
  if (('ontouchstart') in window || (window.DocumentTouch && document instanceof window.DocumentTouch)) {
    return true;
  }

  if (window.matchMedia) {
    const prefixes = ' -webkit- -moz- -o- -ms- '.split(' ');
    const query = ['(', prefixes.join('touch-enabled),('), 'heartz', ')'].join('');
    return window.matchMedia(query).matches;
  }

  return false;
};

class DeveloperSignup extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);

    this.state = {
      error: {
        message: '',
        controls: '',
      },
    };
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit(event) {
    const { username, password, passwordConfirm } = this.state;
    event.preventDefault();

    if (!username) {
      return this.setState({
        error: {
          message: 'Username is required',
          controls: ['username'],
        },
      });
    }

    if (!password) {
      return this.setState({
        error: {
          message: 'Password is required',
          controls: ['password'],
        },
      });
    }

    if (!passwordConfirm) {
      return this.setState({
        error: {
          message: 'Confirmation is required',
          controls: ['confirmation'],
        },
      });
    }

    if (password !== passwordConfirm) {
      return this.setState({
        error: {
          message: 'Password and confirmation do not match',
          controls: ['password', 'confirmation'],
        },
      });
    }

    this.setState({
      loading: true,
      error: {
        message: '',
        controls: '',
      },
    });

    return this.handleStoreLogin().then(() => this.setState({ loading: false }));
  }

  handleStoreLogin() {
    const { username, password } = this.state;
    const storeURL = process.env.REACT_APP_STORE_URL;
    const usersURL = `${storeURL}users/`;
    const authURL = `${storeURL}auth-token/`;
    let authToken;
    let userData;

    return new Promise(async (resolve, reject) => {
      try {
        userData = await StoreClient.createUser(usersURL, username, password, username);
      } catch (e) {
        /* TODO: JCC Enhance error handling */
        this.setState({
          error: {
            message: 'This field must be unique',
            controls: ['username'],
          },
        });
        return resolve(e);
      }
      try {
        authToken = await StoreClient.getAuthToken(authURL, username, password);
      } catch (e) {
        return reject(e);
      }
      this.setState({
        userData,
        authToken,
      });
      return resolve(authToken);
    });
  }

  render() {
    const {
      error, loading, userData, authToken,
    } = this.state;
    const disableControls = loading || userData;
    return (
      <Form onSubmit={this.handleSubmit}>
        <p>{loading ? 'Creating' : 'Create'} a ChRIS Developer account:</p>
        <FormGroup
          controlId="username"
          validationState={error.controls.includes('username') ? 'error' : null}
        >
          <ControlLabel>
            Username
          </ControlLabel>
          <FormControl
            type="text"
            autoComplete="off"
            autoFocus={!isTouchDevice()}
            onChange={this.handleChange}
            name="username"
            disabled={disableControls}
          />
          <HelpBlock>
            { error.controls.includes('username') ? error.message : 'Enter your username' }
          </HelpBlock>
        </FormGroup>
        <FormGroup
          controlId="password"
          validationState={error.controls.includes('password') ? 'error' : null}
        >
          <ControlLabel>
            Password
          </ControlLabel>
          <FormControl
            type="password"
            autoComplete="new-password"
            onChange={this.handleChange}
            name="password"
            disabled={disableControls}
          />
          <HelpBlock>
            { error.controls.includes('password') ? error.message : 'Enter your password' }
          </HelpBlock>
        </FormGroup>
        <FormGroup
          controlId="password-confirm"
          validationState={error.controls.includes('confirmation') ? 'error' : null}
        >
          <ControlLabel>
            Password Confirmation
          </ControlLabel>
          <FormControl
            type="password"
            autoComplete="new-password"
            onChange={this.handleChange}
            name="passwordConfirm"
            disabled={disableControls}
          />
          <HelpBlock>
            { error.controls.includes('confirmation') ? error.message : 'Confirm your password' }
          </HelpBlock>
        </FormGroup>
        <Spinner loading={loading} size="md" inline>
          { userData && authToken ? <Alert type="success"><span>Account created successfully</span></Alert> :
          <Button bsStyle="primary" bsSize="large" type="submit" disabled={disableControls}>
            Create Account
          </Button> }
        </Spinner>{loading && <span className="developer-signup-creating">  Creating Account</span>}
      </Form>);
  }
}

export default DeveloperSignup;
