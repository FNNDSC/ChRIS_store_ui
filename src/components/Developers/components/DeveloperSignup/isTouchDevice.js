import React from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Form,
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock,
  Button,
  Spinner,
} from 'patternfly-react';
import _ from 'lodash';
import StoreClient from '@fnndsc/chrisstoreapi';
import { validate } from 'email-validator';
import './DeveloperSignup.css';
import './DeveloperSignup';
import ChrisStore from '../../../../store/ChrisStore';


const isTouchDevice = () => {
  if (('ontouchstart') in window || (window.DocumentTouch && document instanceof window.DocumentTouch)) {
    return true;
  }

  if (window.matchMedia) {
    const prefixes = ' -webkit- -moz- -o- -ms- '.split(' ');
    const query = ['(', prefixes.join('touch-enabled),('), 'heartz', ')'].join('');
    return window.matchMedia(query).matches;
  }

  return false;
};

export default isTouchDevice;