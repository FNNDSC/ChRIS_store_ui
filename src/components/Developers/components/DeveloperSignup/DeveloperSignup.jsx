import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { validate } from 'email-validator';
import { Form, Spinner } from '@patternfly/react-core';

import './DeveloperSignup.css';
import StoreClient from '@fnndsc/chrisstoreapi';
import Button from '../../../Button';
import ChrisStore from '../../../../store/ChrisStore';
import FormInput from '../../../FormInput';
import isTouchDevice from './isTouchDevice';

export class DeveloperSignup extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    const queryParams = new URLSearchParams(window.location.search);
    const emailVal = queryParams.get('email');
    this.state = {
      loading: false,
      error: {
        message: String(),
        controls: [],
      },
      username: String(),
      email: emailVal,
      password: String(),
      passwordConfirm: String(),
    };
  }

  handleChange(value, name) {
    this.setState({ [name]: value });
  }

  handleSubmit(event) {
    event.preventDefault()
    event.persist();
    const { username, email, password, passwordConfirm } = this.state;
    const { store } = this.props;

    if (!username) {
      return this.setState({
        error: {
          message: 'Username is required',
          controls: ['username'],
        },
      });
    }

    if (!email || !validate(email)) {
      return this.setState({
        error: {
          message: 'A valid Email is required',
          controls: ['email'],
        },
      });
    }

    if (!password) {
      return this.setState({
        error: {
          message: 'Password is required',
          controls: ['password'],
        },
      });
    }

    if (!passwordConfirm) {
      return this.setState({
        error: {
          message: 'Confirmation is required',
          controls: ['confirmation'],
        },
      });
    }

    if (password !== passwordConfirm) {
      return this.setState({
        error: {
          message: 'Password and confirmation do not match',
          controls: ['password', 'confirmation'],
        },
      });
    }

    this.setState(
      {
        loading: true,
        error: {
          message: '',
          controls: '',
        },
      },
      () => store.set('userName')(username)
    );

    return this.handleStoreLogin();
  }

  async handleStoreLogin() {
    const { username, email, password } = this.state;
    const { store } = this.props;
    const storeURL = process.env.REACT_APP_STORE_URL;
    const usersURL = `${storeURL}users/`;
    const authURL = `${storeURL}auth-token/`;
    let authToken;

    try {
      await StoreClient.createUser(usersURL, username, password, email);
    } catch (e) {
      if (_.has(e, 'response')) {
        if (_.has(e, 'response.data.username')) {
          this.setState({
            loading: false,
            error: {
              message: 'This username is already registered.',
              controls: ['username'],
            },
          });
        } else {
          this.setState({
            loading: false,
            error: {
              message: 'This email is already registered.',
              controls: ['email'],
            },
          });
        }
      } else {
        this.setState({
          loading: false,
        });
      }
      return e;
    }

    try {
      authToken = await StoreClient.getAuthToken(authURL, username, password);
    } catch (e) {
      return Promise.reject(e);
    }

    return this.setState(
      {
        toDashboard: true,
      },
      () => {
        store.set('authToken')(authToken);
        return authToken;
      }
    );
  }

  render() {
    const {
      error,
      loading,
      toDashboard,

      username,
      email,
      password,
      passwordConfirm,
    } = this.state;

    if (toDashboard) return <Redirect to="/dashboard" />;

    const disableControls = loading;
    return (
      <Form noValidate id="developer-signup-form" onSubmit={this.handleSubmit}>
        <FormInput
          formLabel="Username"
          fieldId="username"
          validationState={error.controls.includes('username') ? 'error' : 'default'}
          placeholder="Enter your username"
          inputType="text"
          id="username"
          fieldName="username"
          value={username}
          autofocus={!isTouchDevice}
          onChange={(val) => this.handleChange(val, 'username')}
          disableControls={disableControls}
          error={error}
        />
        <FormInput
          formLabel="Email"
          fieldId="email"
          validationState={error.controls.includes('email') ? 'error' : 'default'}
          placeholder="Enter your email"
          inputType="email"
          id="email"
          fieldName="email"
          value={email}
          onChange={(val) => this.handleChange(val, 'email')}
          disableControls={disableControls}
          error={error}
        />
        <FormInput
          formLabel="Password"
          fieldId="password"
          validationState={error.controls.includes('password') ? 'error' : 'default'}
          placeholder="Enter a password"
          inputType="password"
          id="password"
          fieldName="password"
          value={password}
          onChange={(val) => this.handleChange(val, 'password')}
          disableControls={disableControls}
          error={error}
        />
        <FormInput
          formLabel="Password Confirmation"
          fieldId="password-confirm"
          validationState={error.controls.includes('confirmation') ? 'error' : 'default'}
          placeholder="Re-type your password"
          inputType="password"
          id="password-confirm"
          fieldName="passwordConfirm"
          value={passwordConfirm}
          onChange={(val) => this.handleChange(val, 'passwordConfirm')}
          disableControls={disableControls}
          error={error}
        />
        <div style={{ padding: '1em 0' }}>
          {loading ? (
            <Spinner size="md" />
          ) : (
            <Button variant="primary" type="submit" loading={disableControls}>
              Create Account
            </Button>
          )}
          {loading && <span id="developer-signup-creating">Creating Account</span>}
        </div>
      </Form>
    );
  }
}

export default ChrisStore.withStore(DeveloperSignup);

DeveloperSignup.propTypes = {
  store: PropTypes.objectOf(PropTypes.object).isRequired,
};
