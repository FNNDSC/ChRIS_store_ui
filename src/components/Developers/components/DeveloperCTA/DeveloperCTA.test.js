import React from 'react';
import { shallow } from 'enzyme';
import DeveloperCTA from './DeveloperCTA';
import DeveloperSignup from '../DeveloperSignup/DeveloperSignup';

describe('DeveloperCTA', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DeveloperCTA />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render developer-cta div', () => {
    expect(wrapper.find('div.developer-cta')).toHaveLength(1);
  });

  it('should render the Card component', () => {
    expect(wrapper.find('Card')).toHaveLength(1);
  });

  it('should render the CardBody component', () => {
    expect(wrapper.find('CardBody')).toHaveLength(1);
  });

  it('should render the DeveloperSignup component', () => {
    expect(wrapper.containsMatchingElement(<DeveloperSignup />)).toEqual(true);
  });
});
