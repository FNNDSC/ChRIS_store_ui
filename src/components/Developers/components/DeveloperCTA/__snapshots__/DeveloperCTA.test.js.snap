// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeveloperCTA should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeveloperCTA />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row developer-cta-container"
>
        <div
                className="developer-cta-overview"
        >
                <div
                        className="developer-cta-header"
                >
                        Expand the reach of your image processing software
                </div>
                <div
                        className="developer-cta-desc"
                >
                        <p>
                                ChRIS is an 
                                <strong>
                                        open source platform
                                </strong>
                                 for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                <strong>
                                         
                                        common standards, rather than disparate silos
                                </strong>
                                .
                        </p>
                        <p>
                                A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                <strong>
                                        Join us!
                                </strong>
                        </p>
                </div>
        </div>
        <div
                className="developer-cta-form"
        >
                <Card
                        accented={false}
                        aggregated={false}
                        aggregatedMini={false}
                        cardRef={null}
                        className=""
                        matchHeight={false}
                >
                        <CardBody
                                className=""
                        >
                                <DeveloperSignup />
                        </CardBody>
                </Card>
        </div>
</div>,
      "className": "developer-cta",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="developer-cta-overview"
>
            <div
                        className="developer-cta-header"
            >
                        Expand the reach of your image processing software
            </div>
            <div
                        className="developer-cta-desc"
            >
                        <p>
                                    ChRIS is an 
                                    <strong>
                                                open source platform
                                    </strong>
                                     for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                    <strong>
                                                 
                                                common standards, rather than disparate silos
                                    </strong>
                                    .
                        </p>
                        <p>
                                    A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                    <strong>
                                                Join us!
                                    </strong>
                        </p>
            </div>
</div>,
          <div
            className="developer-cta-form"
>
            <Card
                        accented={false}
                        aggregated={false}
                        aggregatedMini={false}
                        cardRef={null}
                        className=""
                        matchHeight={false}
            >
                        <CardBody
                                    className=""
                        >
                                    <DeveloperSignup />
                        </CardBody>
            </Card>
</div>,
        ],
        "className": "row developer-cta-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="developer-cta-header"
>
                Expand the reach of your image processing software
</div>,
              <div
                className="developer-cta-desc"
>
                <p>
                                ChRIS is an 
                                <strong>
                                                open source platform
                                </strong>
                                 for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                <strong>
                                                 
                                                common standards, rather than disparate silos
                                </strong>
                                .
                </p>
                <p>
                                A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                <strong>
                                                Join us!
                                </strong>
                </p>
</div>,
            ],
            "className": "developer-cta-overview",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Expand the reach of your image processing software",
                "className": "developer-cta-header",
              },
              "ref": null,
              "rendered": "Expand the reach of your image processing software",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    ChRIS is an 
                    <strong>
                                        open source platform
                    </strong>
                     for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                    <strong>
                                         
                                        common standards, rather than disparate silos
                    </strong>
                    .
</p>,
                  <p>
                    A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                    <strong>
                                        Join us!
                    </strong>
</p>,
                ],
                "className": "developer-cta-desc",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "ChRIS is an ",
                      <strong>
                        open source platform
</strong>,
                      " for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following",
                      <strong>
                         
                        common standards, rather than disparate silos
</strong>,
                      ".",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "ChRIS is an ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "open source platform",
                      },
                      "ref": null,
                      "rendered": "open source platform",
                      "type": "strong",
                    },
                    " for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          "common standards, rather than disparate silos",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "common standards, rather than disparate silos",
                      ],
                      "type": "strong",
                    },
                    ".",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. ",
                      <strong>
                        Join us!
</strong>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Join us!",
                      },
                      "ref": null,
                      "rendered": "Join us!",
                      "type": "strong",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Card
              accented={false}
              aggregated={false}
              aggregatedMini={false}
              cardRef={null}
              className=""
              matchHeight={false}
>
              <CardBody
                            className=""
              >
                            <DeveloperSignup />
              </CardBody>
</Card>,
            "className": "developer-cta-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "accented": false,
              "aggregated": false,
              "aggregatedMini": false,
              "cardRef": null,
              "children": <CardBody
                className=""
>
                <DeveloperSignup />
</CardBody>,
              "className": "",
              "matchHeight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <DeveloperSignup />,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row developer-cta-container"
>
          <div
                    className="developer-cta-overview"
          >
                    <div
                              className="developer-cta-header"
                    >
                              Expand the reach of your image processing software
                    </div>
                    <div
                              className="developer-cta-desc"
                    >
                              <p>
                                        ChRIS is an 
                                        <strong>
                                                  open source platform
                                        </strong>
                                         for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                        <strong>
                                                   
                                                  common standards, rather than disparate silos
                                        </strong>
                                        .
                              </p>
                              <p>
                                        A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                        <strong>
                                                  Join us!
                                        </strong>
                              </p>
                    </div>
          </div>
          <div
                    className="developer-cta-form"
          >
                    <Card
                              accented={false}
                              aggregated={false}
                              aggregatedMini={false}
                              cardRef={null}
                              className=""
                              matchHeight={false}
                    >
                              <CardBody
                                        className=""
                              >
                                        <DeveloperSignup />
                              </CardBody>
                    </Card>
          </div>
</div>,
        "className": "developer-cta",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="developer-cta-overview"
>
              <div
                            className="developer-cta-header"
              >
                            Expand the reach of your image processing software
              </div>
              <div
                            className="developer-cta-desc"
              >
                            <p>
                                          ChRIS is an 
                                          <strong>
                                                        open source platform
                                          </strong>
                                           for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                          <strong>
                                                         
                                                        common standards, rather than disparate silos
                                          </strong>
                                          .
                            </p>
                            <p>
                                          A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                          <strong>
                                                        Join us!
                                          </strong>
                            </p>
              </div>
</div>,
            <div
              className="developer-cta-form"
>
              <Card
                            accented={false}
                            aggregated={false}
                            aggregatedMini={false}
                            cardRef={null}
                            className=""
                            matchHeight={false}
              >
                            <CardBody
                                          className=""
                            >
                                          <DeveloperSignup />
                            </CardBody>
              </Card>
</div>,
          ],
          "className": "row developer-cta-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="developer-cta-header"
>
                  Expand the reach of your image processing software
</div>,
                <div
                  className="developer-cta-desc"
>
                  <p>
                                    ChRIS is an 
                                    <strong>
                                                      open source platform
                                    </strong>
                                     for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                                    <strong>
                                                       
                                                      common standards, rather than disparate silos
                                    </strong>
                                    .
                  </p>
                  <p>
                                    A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                                    <strong>
                                                      Join us!
                                    </strong>
                  </p>
</div>,
              ],
              "className": "developer-cta-overview",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Expand the reach of your image processing software",
                  "className": "developer-cta-header",
                },
                "ref": null,
                "rendered": "Expand the reach of your image processing software",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      ChRIS is an 
                      <strong>
                                            open source platform
                      </strong>
                       for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following
                      <strong>
                                             
                                            common standards, rather than disparate silos
                      </strong>
                      .
</p>,
                    <p>
                      A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. 
                      <strong>
                                            Join us!
                      </strong>
</p>,
                  ],
                  "className": "developer-cta-desc",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "ChRIS is an ",
                        <strong>
                          open source platform
</strong>,
                        " for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following",
                        <strong>
                           
                          common standards, rather than disparate silos
</strong>,
                        ".",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "ChRIS is an ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "open source platform",
                        },
                        "ref": null,
                        "rendered": "open source platform",
                        "type": "strong",
                      },
                      " for medical analytics in the cloud, democratizing the development of image processing apps within an ecosystem following",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " ",
                            "common standards, rather than disparate silos",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          " ",
                          "common standards, rather than disparate silos",
                        ],
                        "type": "strong",
                      },
                      ".",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. ",
                        <strong>
                          Join us!
</strong>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "A ChRIS Developer account enables you to submit your image processing application as a containerized ChRIS plugin and share it with the broader ChRIS community of researchers and clinicians. ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Join us!",
                        },
                        "ref": null,
                        "rendered": "Join us!",
                        "type": "strong",
                      },
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Card
                accented={false}
                aggregated={false}
                aggregatedMini={false}
                cardRef={null}
                className=""
                matchHeight={false}
>
                <CardBody
                                className=""
                >
                                <DeveloperSignup />
                </CardBody>
</Card>,
              "className": "developer-cta-form",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "accented": false,
                "aggregated": false,
                "aggregatedMini": false,
                "cardRef": null,
                "children": <CardBody
                  className=""
>
                  <DeveloperSignup />
</CardBody>,
                "className": "",
                "matchHeight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <DeveloperSignup />,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
